pandas/core/apply.py

```diff
@@ -465,27 +456,20 @@ def agg_dict_like(self) -> FrameOrSeriesUnion:
 
         return result
 
-    def maybe_apply_str(self) -> FrameOrSeriesUnion | None:
+    def maybe_apply_str(self) -> FrameOrSeriesUnion:
         """
         Compute apply in case of a string.
 
         Returns
         -------
-        result: Series, DataFrame, or None
-            Result when self.f is a string, None otherwise.
+        result: Series or DataFrame
         """
+        # Caller is responsible for checking isinstance(self.f, str)
         f = self.f
-        if not isinstance(f, str):
-            return None
```

Comment:
Not necessarily opposed, but why this change? My original thinking was to have each method determine whether or not it would take action rather than the caller - this way that logic is easily shared between the subclasses.

Reply:
when going through these and seeing

```
result = self.foo()
if result is not None:
    return result

[...]
```

it isnt obvious what cases lead to `result = None`.  e.g. before i checked, i imagined it could have been something like

```
def foo(self):
    try:
        [...]
    except Bar:
        return None
```

Reply:
I see. Looking at this now, I see my usage of `maybe_*` is a bit different than other places in code, where maybe an operation is applied to data but the data is always returned regardless. The downside is that the condition then needs repeated, but it is simple enough (at least at this point) that I don't find that concerning at all.

Reply:
I think this change should be applied to maybe_apply_multiple, and the maybe prefix should be dropped. I'm happy to take care of those in a followup.